### PR TITLE
Adding form validation, and adding enum

### PR DESCRIPTION
**Pull request by T0nio**

### CODE REVIEW

File: src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js [hunk 1]
```diff
@@ -143,8 +162,9 @@ function mapStateToProps(state) {
 
 function mapDispatchToProps(dispatch) {
   return {
-    addActivity: (activity) => dispatch(addActivity(activity))
-  }
+    addActivity: (activity) => dispatch(addActivity(activity)),
+    addInterlocutor: (interlocutor) => dispatch(addInterlocutor(interlocutor))
+  };
```
Comment (yoagauthier) [inline src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js]: Why addInterlocutor is here ? Should not be in that PR ...

File: src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js [hunk 2]
```diff
@@ -6,23 +6,28 @@ moment.locale('fr');
...
-      key: this.props.lastID + 1
+      activity: {
+        title: "",
+        description: "",
+        date: moment(),
+        type: ActivityTypeEnum.CONTENT,
+        contentSource: "",
+        eventWhat: "",
+        meetingWho: "",
+        tags: [],
```
Comment (yoagauthier) [inline src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js]: eventWhat and meetingWho correspond to which Database fields ? Maybe thay should also be renamed right ?

File: src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js [hunk 3]
```diff
@@ -6,23 +6,28 @@ moment.locale('fr');
...
+        key: this.props.lastID + 1,
+      },
```
Comment (yoagauthier) [inline src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js]: I think we should also rename "key" into "id" if possible to ensure coherence with the DB.

**Review by yoagauthier — DISMISSED** [review]

File: src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js (hunk 2 above)
Comment (T0nio) [reply src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js]: Done

File: src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js (hunk 3 above)
Comment (T0nio) [reply src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js]: It'll be fixed in the next pr by @sioufia

File: src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js (hunk 1 above)
Comment (T0nio) [reply src/components/activity/screens/ActivityAdd/containers/ActivityAddContainer.js]: Fixed

**Pull request merged**